File: 剑指offer/57. 和为s的两个数字/57. 和为s的两个数字.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i=0;
        int j=nums.size()-1;
        while((nums[i]+nums[j])!=target){
            if((nums[i]+nums[j])>target){
                j--;
            }
            else i++;
        }
        return {nums[i],nums[j]};
    }
};

File: 剑指offer/57. 和为s的两个数字/57. 和为s的两个数字.py
# 双指针


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        i,j=0,len(nums)-1
        while nums[i]+nums[j]!=target:
            if nums[i]+nums[j]>target:
                j-=1
            else:
                i+=1
        return [nums[i],nums[j]]